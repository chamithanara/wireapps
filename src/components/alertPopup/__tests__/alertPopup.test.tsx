import { Alert, Platform } from 'react-native';
import AlertPopupComponent from '../index';

describe('Testing AlertPopup Component', () => {
    let onOkActionPressMock: jest.Mock;
    let onCancelActionPressMock: jest.Mock;

    beforeAll(() => {
        onOkActionPressMock = jest.fn();
        onCancelActionPressMock = jest.fn();
    });

    describe('Rendering alert - ios', () => {
        beforeEach(() => {
            Platform.OS = 'ios';
        });

        test('Check text', () => {
            Platform.OS = 'ios';
            const mockedAlertPopup = jest.fn();
            jest.spyOn(Alert, 'alert').mockImplementation(mockedAlertPopup);

            AlertPopupComponent.showAlert({
                message: 'message',
                cancelLabel: 'cancel',
                okLabel: 'ok',
                title: 'title',
                onOkAction: onOkActionPressMock,
                onCancelAction: onCancelActionPressMock
            });

            expect(mockedAlertPopup).toHaveBeenCalledTimes(1);
        });
    });

    describe('Alert Tests', () => {
        test('check title with ok button', () => {
            const mockedPopup = jest.fn();
            jest.spyOn(AlertPopupComponent, 'showAlert').mockImplementation(mockedPopup);

            AlertPopupComponent.showAlert({
                message: 'message',
                okLabel: 'Ok',
                title: 'title',
                onOkAction: onOkActionPressMock
            });

            expect(mockedPopup).toHaveBeenCalledTimes(1);
            expect(mockedPopup).toMatchSnapshot();
        });

        test('check title with cancel button', () => {
            const mockedPopup = jest.fn();
            jest.spyOn(AlertPopupComponent, 'showAlert').mockImplementation(mockedPopup);

            AlertPopupComponent.showAlert({
                message: 'message',
                cancelLabel: 'cancel',
                title: 'title',
                onCancelAction: onCancelActionPressMock
            });

            expect(mockedPopup).toHaveBeenCalledTimes(1);
            expect(mockedPopup).toMatchSnapshot();
        });

        test('check title with two buttons', () => {
            const mockedPopup = jest.fn();
            jest.spyOn(AlertPopupComponent, 'showAlert').mockImplementation(mockedPopup);

            AlertPopupComponent.showAlert({
                message: 'message',
                cancelLabel: 'cancel',
                okLabel: 'Ok',
                title: 'title',
                onOkAction: onOkActionPressMock,
                onCancelAction: onCancelActionPressMock
            });

            expect(mockedPopup).toHaveBeenCalledTimes(1);
            expect(mockedPopup).toMatchSnapshot();
        });

        test('check outsideDismiss true', () => {
            const mockedPopup = jest.fn();
            jest.spyOn(AlertPopupComponent, 'showAlert').mockImplementation(mockedPopup);

            AlertPopupComponent.showAlert({
                message: 'message',
                cancelLabel: 'cancel',
                okLabel: 'Ok',
                title: 'title',
                outsideDismiss: true,
                onOkAction: onOkActionPressMock,
                onCancelAction: onCancelActionPressMock
            });

            expect(mockedPopup).toHaveBeenCalledTimes(1);
            expect(mockedPopup).toMatchSnapshot();
        });
    });
});
