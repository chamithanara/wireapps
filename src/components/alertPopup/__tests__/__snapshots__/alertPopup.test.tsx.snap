// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AlertPopup Component Alert Tests check outsideDismiss true 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "cancelLabel": "cancel",
        "message": "message",
        "okLabel": "Ok",
        "onCancelAction": [MockFunction],
        "onOkAction": [MockFunction],
        "outsideDismiss": true,
        "title": "title",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Testing AlertPopup Component Alert Tests check title with cancel button 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "cancelLabel": "cancel",
        "message": "message",
        "onCancelAction": [MockFunction],
        "title": "title",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Testing AlertPopup Component Alert Tests check title with ok button 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "message": "message",
        "okLabel": "Ok",
        "onOkAction": [MockFunction],
        "title": "title",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Testing AlertPopup Component Alert Tests check title with two buttons 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "cancelLabel": "cancel",
        "message": "message",
        "okLabel": "Ok",
        "onCancelAction": [MockFunction],
        "onOkAction": [MockFunction],
        "title": "title",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
